### PR TITLE
Various error-handling fixes

### PR DESCRIPTION
**Pull request by jlebon**
See individual commit messages.

### CODE REVIEW

**Comment by jlebon** [comment]
Marked as WIP for now since this does not yet address (3) in https://github.com/openshift/machine-config-operator/issues/199#issuecomment-443207455.

**Comment by ashcrow** [comment]
Changes look good. I'm not crazy about `daemon: retry for 5 minutes to get MachineConfig ` though. Is that more so related to https://github.com/openshift/machine-config-operator/issues/203?

**Comment by ashcrow** [comment]
These commits are currently being manually tested by @kikisdeliveryservice and @jlebon

**Comment by jlebon** [comment]
OK, dropping WIP on this, though I'd appreciate some manual testing from others. I pushed another commit which reworks how we use channels and error-handling in goroutines. This allows us to avoid the nuclear reboot-on-error pattern in favour of the mark-degraded-and-exit-on-error pattern.

File: pkg/daemon/writer.go [hunk 1]
```diff
@@ -78,8 +79,10 @@ func (nw *NodeWriter) SetUpdateWorking(client corev1.NodeInterface, node string)
 	return <-respChan
 }
 
-// SetUpdateDegraded Sets the state to UpdateDegraded.
-func (nw *NodeWriter) SetUpdateDegraded(client corev1.NodeInterface, node string) error {
+// SetUpdateDegraded logs the error and sets the state to UpdateDegraded.
+// Returns an error if it couldn't set the annotation.
+func (nw *NodeWriter) SetUpdateDegraded(err error, client corev1.NodeInterface, node string) error {
+	glog.Error("Marking degraded due to: %v", err)
```
Comment (kikisdeliveryservice) [inline pkg/daemon/writer.go]: Think this should be `glog.Errorf` because I'm seeing `Marking degraded due to: %vdaemon can't reconcile this config` :)

Comment (jlebon) [reply pkg/daemon/writer.go]: Thanks, fixed!

**Pull request merged**